app/models/concerns/presenters/creation_behaviour.rb

```diff
@@ -15,6 +15,10 @@ def compatible_tube_purposes
     construct_buttons(purposes_of_type('tube'))
   end
 
+  def compatible_tube_rack_purposes
+    construct_buttons(purposes_of_type('tube_rack'))
+  end
+
```

Comment:
Feels like this wants to be refactored to:
```
def compatible_labware_purposes(labware_type)
    construct_buttons(purposes_of_type(labware_type))
end
```

But again, maybe we can save up refactoring ideas for another story to try to draw a line under this story! Will keep commenting them if I notice and we can decide what to do.

Reply:
Yeah. I think this is building the 'other actions' grey buttons. That could do with some better ordering/structuring in any case.